test_scripts/LowVoltage/001_Low_Voltage_and_Wake_Up.lua

```diff
@@ -0,0 +1,253 @@
+---------------------------------------------------------------------------------------------------
+--
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/LowVoltage/common')
+local runner = require('user_modules/script_runner')
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local hashId = { }
+local grammarId = { }
+
+--[[ Local Functions ]]
+local function checkSDLIgnoresRPCFromMobileSide()
+  common.getMobileSession():SendRPC("AddCommand", { cmdID = 2, vrCommands = { "OnlyVRCommand" }})
+
+  common.getHMIConnection():ExpectAny():Times(0)
+  :Do(function(_, data)
+      print("HMI Event")
+      commonFunctions:printTable(data)
+    end)
+
+  common.getMobileSession():ExpectAny():Times(0)
+  :Do(function(_, data)
+      print("Mobile Event")
+      commonFunctions:printTable(data)
+    end)
+  common.delayedExp(11000)
+end
+
+local function checkSDLIgnoresRPCFromHMISide()
+  common.getHMIConnection():SendNotification("BasicCommunication.OnEventChanged", {
+    isActive = true, eventName = "EMERGENCY_EVENT" })
+
+  common.getHMIConnection():ExpectAny():Times(0)
+  :Do(function(_, data)
+      print("HMI Event")
+      commonFunctions:printTable(data)
+    end)
+
+  common.getMobileSession():ExpectAny():Times(0)
+  :Do(function(_, data)
+      print("Mobile Event")
+      commonFunctions:printTable(data)
+    end)
+  common.delayedExp(11000)
+end
+
+local function configureHMILevels(pNumOfApps)
+  local apps = {
+    [1] = { 1 },
+    [2] = { 2, 1 },
+    [3] = { 2, 3, 1 },
+    [4] = { 2, 3, 1 }
+  }
+  for k, i in pairs(apps[pNumOfApps]) do
+    local function activateApp()
+      local cid = common.getHMIConnection():SendRequest("SDL.ActivateApp", { appID = common.getHMIAppId(i) })
+      common.getHMIConnection():ExpectResponse(cid)
+      :Do(function() common.cprint(35, "Activate App: " .. i) end)
+    end
+    RUN_AFTER(activateApp, 100 * k)
+  end
+  common.getMobileSession(1):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "FULL", systemContext = "MAIN", audioStreamingState = "NOT_AUDIBLE" })
+
+  if pNumOfApps >= 2 then
+    common.getMobileSession(2):ExpectNotification("OnHMIStatus",
+      { hmiLevel = "FULL", systemContext = "MAIN", audioStreamingState = "AUDIBLE" },
+      { hmiLevel = "LIMITED", systemContext = "MAIN", audioStreamingState = "AUDIBLE" })
+    :Times(2)
+  end
+  if pNumOfApps >= 3 then
+    common.getMobileSession(3):ExpectNotification("OnHMIStatus",
+      { hmiLevel = "FULL", systemContext = "MAIN", audioStreamingState = "NOT_AUDIBLE" },
+      { hmiLevel = "BACKGROUND", systemContext = "MAIN", audioStreamingState = "NOT_AUDIBLE" })
+    :Times(2)
+  end
+end
+
+local function addResumptionData(pAppId)
+  local f = {}
+  f[1] = function()
```

Comment:
@dboltovskyi I would propose to create collection of functions which consume application id as parameter to avoid hardcode and improve flexibility of code.

Example:
```
local f = {}
f.AddCommand = function(pAppId)
   local cid = common.getMobileSession(pAppId):SendRPC("AddCommand", { cmdID = 1, vrCommands = { "OnlyVRCommand" }})
    common.getHMIConnection():ExpectRequest("VR.AddCommand")
    :Do(function(_, data)
        grammarId[1] = data.params.grammarID
        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
      end)
    common.getMobileSession(pAppId):ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
    common.getMobileSession(pAppId):ExpectNotification("OnHashChange")
    :Do(function(_, data)
        hashId[1] = data.payload.hashID
      end)
end

f.AddSubMenu = function(pAppId)
  ...
end

f.CreateInteractionChoiceSet = function(pAppId)
  ...
end

f.NoRPC = function(pAppId)   end
  ...
end

local rpc = ... -- logic of choice RPC by application id
f[rpc](pAppId)
```

Reply:
@aderiabin Accepted. Please check af2e393